### PR TITLE
Ensure ProcessModule for main executable is first in modules list

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/24070

@janvorli, can you verify this fixes the issue you were seeing on Alpine in docker?

### CODE REVIEW

**Comment by janvorli** [comment]
@stephentoub I can confirm that it fixes the issue on Alpine

File: src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs [hunk 1]
```diff
@@ -80,6 +80,22 @@ internal static ProcessModuleCollection GetModules(int processId)
                 }
             }
 
+            // Move the main executable module to be the first in the list if it's not already
+            string exePath = Process.GetExePath(processId);
```
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs]: That causes complications if the module is split across multiple entries. And in the vast majority of cases (all we've seen thus far until Alpine in docker), the extra loop will just check the first entry and exit as the main module will already be first.

**Comment by stephentoub** [comment]
> I can confirm that it fixes the issue on Alpine

Thanks for checking.

File: src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (janvorli) [reply src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Linux.cs]: Ah, right, I've forgotten about that.

**Pull request merged**